lib/rules/define-macros-order.js

```diff
@@ -247,17 +246,28 @@ function create(context) {
   }
 
   /**
-   * @param {ASTNode} tokenBefore
-   * @param {ASTNode} tokenAfter
+   * Get result text to insert
+   * @param {string} textNode
+   * @param {ASTNode} target
    */
-  function getTextBetweenTokens(tokenBefore, tokenAfter) {
-    return sourceCode.text.slice(tokenBefore.range[1], tokenAfter.range[0])
+  function getInsertText(textNode, target) {
+    const afterTargetComment = sourceCode.getTokenAfter(target, {
+      includeComments: true
+    })
+    const afterText = sourceCode.text.slice(
+      target.range[1],
+      afterTargetComment.range[0]
+    )
+    // handle case when a();b() -> b()a();
+    const invalidResult = !textNode.endsWith(';') && !afterText.includes('\n')
```

Comment:
This is corner case `a();b() -> b()a();`.
`afterText` contains symbols between target token(`a();`) and next token(`b()`). If there are no new lines and `b()` doesn't ends with `;` - we add `;` manually

Reply:
Can you please add a test case for this edge case?

Reply:
Sure. Done